pkg/provisioner/ironic/ironic.go

```diff
@@ -1219,6 +1219,31 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 		return p.setMaintenanceFlag(ironicNode, true)
 	}
 
+	p.log.Info("requesting ports to delete")
+	port_list, err = ports.List(
+		p.client,
+		ports.ListOpts{NodeUUID: p.status.ID},
+	).Extract()
+	switch err.(type) {
```

Comment:
This switch statement can be replaced with

```
if err != nil {
  return result, errors.Wrap(err, "failed to obtain port list")
}
```

Reply:
I've floated the idea that we likely should have a switch for it. I think there was an original case where there was a case to keep it, but I think that is long past. Additionally, the solution is never supposed to be actually removing the node from ironic, even though we seem to teach that pattern in some projects :(